R/annotation-ticks.R

```diff
@@ -0,0 +1,351 @@
+#'
+#' #add minor ticks on both scales
+#' p1 + annotation_ticks(sides='lb',scale=c('identity','log10'))
+#'
+#' #add minor ticks on both scales, but force x axis to be identity
+#' p1+annotation_ticks(sides='lb',scale='identity')
+#'
+#'
+#' #log scale
+#' p2 <- p + scale_x_continuous(trans='log')
+#'
+#' #add minor ticks on log scale
+#' p2 + annotation_ticks(sides='b',scale='log')
+#'
+#' #add minor ticks on both scales
+#' p2 + annotation_ticks(sides='lb',scale=c('identity','log'))
+#'
+#' #add minor ticks on both scales, but force x axis to be identity
+#' p2 + annotation_ticks(sides='lb',scale='identity')
+#'
+#' @import grid
+annotation_ticks <- function(sides = "b",
+                             scale = 'identity',
+                             scaled = TRUE,
+                             short = grid::unit(0.1,"cm"),
+                             mid = grid::unit(0.2, "cm"),
+                             long = grid::unit(0.3, "cm"),
+                             colour = "black",
+                             size = 0.5,
+                             linetype = 1,
+                             alpha = 1,
+                             color = NULL,
+                             ticks_per_base = NULL,
+                             ...) {
+
+    if (!is.null(color))
+        colour <- color
+
+    # if(!'scale'%in%names(match.call())){
+    #
+    #   scale <- sapply(sides,function(x){
+    #     if(grepl('[b|t]',x))
+    #
+    #          # is there a way to learn what trans is set to at this stage
+    #          # from the object annotation is be applied to?
+    #          # if so then user wont need to supply param to specify non identity scale
+    #
+    #   })
+    #
+    # }
+
+    #check for invalid side
+    if(nchar(gsub('[b|t|l|r]','',sides))>0)
+      stop(gsub('[b|t|l|r]','',sides),' is not a valid side: b,t,l,r are valid')
+
+    #split sides to character vector
+    sides <- strsplit(sides,'')[[1]]
+
+    if((length(sides)!=length(scale))){
+
+      if(length(scale)==1){
+
+        scale <- rep(scale,length(sides))
+
+      }else{
+
+        stop('Number of scales does not match the number of sides')
+
+      }
+
+    }
+
+    base <- sapply(scale,function(x) switch(x,'identity'=10,'log10'=10,'log'=exp(1)),USE.NAMES = FALSE)
+
+    if(!'ticks_per_base'%in%names(match.call())){
+
+      ticks_per_base <- base - 1
+
+    }else{
+
+      if((length(sides)!=length(ticks_per_base))){
+
+        if(length(ticks_per_base)==1){
+
+          ticks_per_base <- rep(ticks_per_base,length(sides))
+
+        }else{
+
+          stop('Number of ticks_per_base does not match the number of sides')
+
+        }
+
+      }
+    }
+
+    delog<-scale%in%'identity'
+
+
+    layer(
+      data = data.frame(x = NA),
+      mapping = NULL,
+      stat = StatIdentity,
+      geom = GeomTicks,
+      position = PositionIdentity,
+      show.legend = FALSE,
+      inherit.aes = FALSE,
+      params = list(base = base,
```

Comment:
`base` should be on a new line as in [here](https://github.com/tidyverse/ggplot2/blob/master/R/annotation-logticks.r#L92)